vars/provisionNodes.groovy

```diff
@@ -163,239 +126,180 @@ def call(Map config = [:]) {
+  new_config['post_reboot'] = cleanup_logs
 
-  sshagent (credentials: ['daos-provisioner']) {
-    // Prefer to attempt delete these logs before the reboot / re-image attempt
-    // so that we have some record.
-    def ns_rc = 0
-    ns_rc = sh script: cleanup_logs,
-               label: "Clean up previous run's logs",
-               returnStatus: true
-    if (ns_rc != 0) {
-      println("Failed to remove previous runs log file(s), etc.")
-    }
-    println "Previous run's log cleanup result = ${ns_rc}"
-    if (config['power_only']) {
-      sh script:'clush -l root -w ' + nodeString +
-                ' --connect_timeout 30 --command_timeout 120 -S' +
-                ' "sync;sync"',
-         label: "Syncing disks",
-         returnStatus: true
-      ns_rc = sh script: """./jenkins/node_powercycle.py \
-                            --node=${nodeString}""",
-                 label: "Power-cycling ${nodeString}",
-                 returnStatus: true
-      println "Powercycle result = ${ns_rc}"
-    } else {
-      ns_rc = sh script: """./jenkins/node_provision_start.py \
-                            --nodes=${nodeString} ${options}""",
-                 label: "Provisioning nodes ${nodeString}",
-                 returnStatus: true
-      println "Snapshot restore result = ${ns_rc}"
-    }
-
-    def woptions = ''
-    if (wait_for_it) {
-      if (config['timeout']) {
-        if (config['timeout'] <= 0) {
-          wait_for_it = false
-        } else {
-          woptions = " --timeout=${config['timeout']}"
-        }
-      }
-      if (config['poll']) {
-        woptions = " --poll=${config['poll']}"
-      }
-    }
-    if (wait_for_it) {
-      def rc = 0
-      rc = sh script: """./jenkins/wait_for_node_ready.py \
-                       --nodes=${nodeString} ${woptions}""",
-                 label: "Waiting for nodes ${nodeString} to be ready",
-              returnStatus: true
-      if (rc != 0) {
-        error "One or more nodes failed to provision!"
-      }
-      ns_rc = sh script: cleanup_logs,
-                 label: "Clean up previous run's logs",
-                 returnStatus: true
-      if (ns_rc != 0) {
-        // If a powercycle fails to clean things up fail the stage.
-        error("Failed to remove pre-existing /tmp/server.log file(s), etc.")
-      }
-
-    }
-
-    // Prepare the node for daos/cart testing
-    def provision_script = '''set -ex
-                              rm -f ci_key*
-                              ssh-keygen -N "" -f ci_key
-                              cat << "EOF" > ci_key_ssh_config
+  // Prepare the node for daos/cart testing
+  def provision_script = '''set -ex
+                            rm -f ci_key*
+                            ssh-keygen -N "" -f ci_key
+                            cat << "EOF" > ci_key_ssh_config
 host wolf-*
     CheckHostIp no
     StrictHostKeyChecking no
     UserKnownHostsFile /dev/null
     LogLevel error
 EOF'''
```

Comment:
This **is** something that should go into a system-pipeline-lib, or some other _site-setting_.

Not suggesting it needs to be fixed here.  Just sayin'.